Sources/ProcedureKit/Group.swift

```diff
@@ -156,10 +156,7 @@ open class GroupProcedure: Procedure, ProcedureQueueDelegate {
         // we must cancelAllOperations and also ensure the queue is not suspended.
         queue.cancelAllOperations()
         queue.isSuspended = false
-
-        // If you find that execution is stuck on the following line, one of the child
-        // Operations/Procedures is likely not handling cancellation and finishing.
-        queue.waitUntilAllOperationsAreFinished()
```

Comment:
To add-on (and for future reference): There looks to be some difference in how things are being scheduled / pre-empted on different machines, as:
- The machine with 4 cores + Hyper-Threading **never** encountered this particular order of operations that could result in deadlock.
- The machine with 4 cores (but *no* Hyper-Threading) **consistently encountered** this particular order of operations that could result in deadlock (i.e. every time a particular Stress Test was run).
 > Overall failures only occurred in the second batch of `test__group_cancel()`, and often around 2/3 of the second batch of Group's would fail to finish.